source/fe/fe_system.cc

```diff
@@ -2009,6 +2009,69 @@ FESystem<dim, spacedim>::initialize(
       Assert(index == this->n_dofs_per_line(), ExcInternalError());
     });
 
+
+  // Compute local_dof_sparsity_pattern if any of our base elements contains a
+  // non-empty one (empty denotes the default of all DoFs coupling within a
+  // cell). Note the we currently only handle coupling within a base element and
+  // not between two different base elements. Handling the latter could be
+  // doable if the underlying element happens to be identical, but we currently
+  // have no functionality to compute the coupling between different elements
+  // with a pattern (for example FE_Q_iso_Q1 with different degrees).
+  {
+    // Does any of our base elements not couple all DoFs?
+    const bool have_nonempty = [&]() -> bool {
+      for (unsigned int b = 0; b < this->n_base_elements(); ++b)
+        {
+          if (!this->base_element(b).get_local_dof_sparsity_pattern().empty() &&
+              (this->element_multiplicity(b) > 0))
+            return true;
+        }
+      return false;
+    }();
+
+    if (have_nonempty)
+      {
+        this->local_dof_sparsity_pattern.reinit(this->n_dofs_per_cell(),
+                                                this->n_dofs_per_cell());
+        // by default, everything couples:
+        this->local_dof_sparsity_pattern.fill(true);
+
+        unsigned int current_dof_index = 0;
+
+        for (unsigned int b = 0; b < this->n_base_elements(); ++b)
+          {
+            const auto &pattern =
+              this->base_element(b).get_local_dof_sparsity_pattern();
+            const unsigned int m = this->element_multiplicity(b);
```

Comment:
I would find the loops below easier to read if this were given a more descriptive name, e.g.
```suggestion
            const unsigned int multiplicity = this->element_multiplicity(b);
```